src/CqlBuilderPanel/Parameters/ParameterBuilder.tsx

```diff
@@ -0,0 +1,158 @@
+              tw="w-full"
+              readOnly={!canEdit}
+              disabled={!canEdit}
+              label="Paramter Name"
+              placeholder=""
+              inputProps={{
+                "data-testid": "parameter-name-text-input",
+              }}
+              {...formik.getFieldProps("parameterName")}
+              error={Boolean(formik.errors.parameterName)}
+              helperText={formik.errors.parameterName}
+            />
+          </div>
+        </div>
+        <br />
+
+        <FormikProvider value={formik}>
+          <AceEditor
+            mode="sql"
+            ref={textAreaRef}
+            theme="monokai"
+            value={formik.values.expression}
+            onChange={(value) => {
+              formik.setFieldValue("expression", value);
+            }}
+            onLoad={(aceEditor) => {
+              // On load we want to tell the ace editor that it's inside of a scrollabel page
+              aceEditor.setOption("autoScrollEditorIntoView", true);
+            }}
+            width="100%"
+            height={editorHeight}
+            wrapEnabled={true}
+            readOnly={false}
+            name="ace-editor-wrapper"
+            enableBasicAutocompletion={true}
+            //@ts-ignore
+          />
+        </FormikProvider>
+        <div style={{ marginTop: "24px" }}>
+          <div className="form-actions">
+            <Button
+              variant="outline"
+              data-testid="parameter-cancel-btn"
+              disabled={
+                (!formik.dirty &&
+                  expressionEditorValue ===
+                    (parameter?.expressionValue || "")) ||
+                !canEdit
+              }
+              tw="mr-4"
+              onClick={() => {
+                resetForm();
+                setOpenParameterDialog(false);
+              }}
+            >
+              Cancel
+            </Button>
+            <Button
+              data-testid={`parameter-${
+                operation === "edit" ? "save" : "apply"
+              }-btn`}
+              disabled={
+                !formik.values.parameterName ||
+                !canEdit ||
+                !expressionEditorValue ||
+                isEditDialogFormDirty() ||
+                !formik.isValid
+              }
+              onClick={() => {
+                const parameterToApply: Parameter = {
+                  parameterName: formik.values.parameterName,
+                  expressionValue: expressionEditorValue,
+                };
+                resetForm();
+                setExpressionEditorValue("");
+                if (operation === "edit") {
+                  formik.setFieldValue("parameterName", "");
+                  // call handleParameterEdit
+                  onClose();
+                } else {
+                  handleApplyParameter(parameterToApply);
+                }
+              }}
+            >
+              {operation === "edit" ? "Save" : "Apply"}
```

Comment:
I only see this being used for edit. is this condition required? if yes, can we use this component for apply action as well by deleting the `ParameterPane.tsx`? both look quite similar

Reply:
probably we can add a tech debt to refactor this. I think we can reuse the new parameter and edit parameter forms. rest looks good.